pkg/nfd-topology-updater/kubeletnotifier/kubeletnotifier.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeletnotifier
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+type EventType string
+
+const (
+	IntervalBased EventType = "intervalBased"
+	FSUpdate      EventType = "fsUpdate"
+)
+
+var stateFiles = sets.NewString(
+	"cpu_manager_state",
+	"memory_manager_state",
+	"kubelet_internal_checkpoint",
+)
+
+type Notifier struct {
+	sleepInterval time.Duration
+	// destination where notifications are sent
+	dest    chan<- Info
+	fsEvent <-chan fsnotify.Event
+}
+
+type Info struct {
+	Event EventType
+}
+
+func New(sleepInterval time.Duration, dest chan<- Info, kubeletDirPath string) (*Notifier, error) {
+	ch, err := createFSWatcherEvent([]string{kubeletDirPath})
+	if err != nil {
+		return nil, err
+	}
+	return &Notifier{
+		sleepInterval: sleepInterval,
+		dest:          dest,
+		fsEvent:       ch,
+	}, nil
+}
+
+func (n *Notifier) Run() {
+	timeEvents := make(<-chan time.Time)
+	if n.sleepInterval > 0 {
+		ticker := time.NewTicker(n.sleepInterval)
+		timeEvents = ticker.C
+	}
+
+	for {
+		select {
+		case <-timeEvents:
+			klog.V(5).Infof("timer update received")
+			i := Info{Event: IntervalBased}
+			n.dest <- i
+
+		case e := <-n.fsEvent:
+			klog.V(5).Infof("fsnotify event from file %q: %q received", e.Name, e.Op)
+			if stateFiles.Has(e.Name) {
+				i := Info{Event: FSUpdate}
+				n.dest <- i
+			}
+		}
+	}
+}
+
+func createFSWatcherEvent(fsWatchPaths []string) (chan fsnotify.Event, error) {
+	es := make(chan fsnotify.Event)
+	fsWatcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return es, err
+	}
+	for _, path := range fsWatchPaths {
+		if err = fsWatcher.Add(path); err != nil {
+			return es, fmt.Errorf("failed to watch: %q; %w", path, err)
+		}
+	}
+	es = fsWatcher.Events
+	return es, nil
+}
```

Comment:
small nit, i think it would be easier to read it like this:
```suggestion
func createFSWatcherEvent(fsWatchPaths []string) (chan fsnotify.Event, error) {
	fsWatcher, err := fsnotify.NewWatcher()
	if err != nil {
		return nil, err
	}
	for _, path := range fsWatchPaths {
		if err = fsWatcher.Add(path); err != nil {
			return nil, fmt.Errorf("failed to watch: %q; %w", path, err)
		}
	}
	return fsWatcher.Events, nil
}
```
this one is non blocking,  LGTM from me

Reply:
Done. Thanks!